auction/manager.go

```diff
@@ -171,6 +173,17 @@ func (am *Manager) NewSolverOperation(solverOp *operation.SolverOperation) *rela
 		return ErrNotEnoughBondedBalance
 	}
 
+	solverGasLimit, err := SolverGasLimit(solverOp.Control, am.ethClient)
+	if err != nil {
+		log.Info("failed to get solver gas limit", "err", err, "userOpHash", solverOp.UserOpHash.Hex())
+		return relayerror.ErrServerInternal
+	}
+
+	if solverOp.Gas.Cmp(big.NewInt(int64(solverGasLimit))) > 0 {
+		log.Info("solver operation gas is more than solver gas limit", "userOpHash", solverOp.UserOpHash.Hex(), "solverGasLimit", solverGasLimit, "solverOpGas", solverOp.Gas.String())
+		return ErrInvalidSolverGas
+	}
+
```

Comment:
Feed `solverGasLimit` to `solverOp.Validate()` above instead. Then you don't need the new error (4005).

Reply:
Its here because it requires an ethclient call

Reply:
done